packages/@glimmer/tracking/src/cached.ts

```diff
@@ -0,0 +1,108 @@
+import { DEBUG } from '@glimmer/env';
+import { createCache, getValue } from '@glimmer/validator';
+
+/**
+ * @decorator
+ *
+ * The `@cached` decorator can be used on getters in order to cache the return
+ * value of the getter. This is useful when a getter is expensive and used very
+ * often.
+ *
+ *
+ * @example
+ *
+ * in this guest list class, we have the `sortedGuests`
+ * getter that sorts the guests alphabetically:
+ *
+ * ```js
+ * import { tracked } from '@glimmer/tracking';
+ * class GuestList {
+ *   @tracked guests = ['Zoey', 'Tomster'];
+ *   get sortedGuests() {
+ *     return this.guests.slice().sort()
+ *   }
+ * }
```

Comment:
I'd recommend being generous with empty lines, since this will be read a bit. So a new-line after the import, and one after the guests definition.